src/wallet/wallet.h

```diff
@@ -339,6 +339,7 @@ class CWalletTx : public CMerkleTx
     char fFromMe;
     std::string strFromAccount;
     int64_t nOrderPos; //!< position in ordered transaction list
+    std::multimap<int64_t, std::pair<CWalletTx*, CAccountingEntry*>>::const_iterator m_it_wtxOrdered;
```

Comment:
Remove, `nOrderPos` should be enough for fast lookup in `wtxOrdered`.

Reply:
@promag IIRC, `nOrderPos` is updated while the cache keeps the old value...

If I do this, I'd have to open a new pull request, since this is a different fix.

Reply:
Either way storing a map/multimap iterator is safe (while the entry exists) and this is the most performant solution AFAIK.